modules/collect/src/main/java/com/opengamma/strata/collect/io/XmlElement.java

```diff
@@ -0,0 +1,391 @@
+
+/**
+ * A single element in the tree structure of XML.
+ * <p>
+ * This class is a minimal, lightweight representation of an element in the XML tree.
+ * The element has a name, attributes, and either content or children.
+ * <p>
+ * Note that this representation does not express all XML features.
+ * No support is provided for processing instructions, comments or mixed content.
+ * In addition, it is not possible to determine the difference between empty content and no children.
+ * <p>
+ * There is no explicit support for namespaces.
+ * When creating instances, the caller may choose to use a convention to represent namespaces.
+ * For example, element and attribute names may use prefixes or the standard {@link QName} format.
+ */
+public final class XmlElement
+    implements ImmutableBean {
+
+  /**
+   * The meta-bean.
+   * This is a manually coded bean.
+   */
+  private static MetaBean META_BEAN = LightMetaBean.of(XmlElement.class);
+
+  /**
+   * The element name.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final String name;
+  /**
+   * The attributes.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ImmutableMap<String, String> attributes;
+  /**
+   * The element content.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final String content;
+  /**
+   * The child nodes.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ImmutableList<XmlElement> children;
+
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains an instance with content and no attributes.
+   * <p>
+   * Returns an element representing XML with content, but no children.
+   * 
+   * @param name  the element name, not empty
+   * @param content  the content, empty if the element has no content
+   * @return the element
+   */
+  public static XmlElement ofContent(String name, String content) {
+    return ofContent(name, ImmutableMap.of(), content);
+  }
+
+  /**
+   * Obtains an instance with content and attributes.
+   * <p>
+   * Returns an element representing XML with content and attributes but no children.
+   * 
+   * @param name  the element name, not empty
+   * @param attributes  the attributes, empty if the element has no attributes
+   * @param content  the content, empty if the element has no content
+   * @return the element
+   */
+  public static XmlElement ofContent(String name, Map<String, String> attributes, String content) {
+    return new XmlElement(name, ImmutableMap.copyOf(attributes), content, ImmutableList.of());
+  }
+
+  /**
+   * Obtains an instance with children and no attributes.
+   * <p>
+   * Returns an element representing XML with children, but no content.
+   * 
+   * @param name  the element name, not empty
+   * @param children  the children, empty if the element has no children
+   * @return the element
+   */
+  public static XmlElement ofChildren(String name, List<XmlElement> children) {
```

Comment:
Is it worth having overloads for this method and the one below where the children are varags? That would provide a cleaner way to define an XML structure in code.